src/script/lua_api/l_client.cpp

```diff
@@ -183,6 +193,10 @@ int ModApiClient::l_get_node_or_nil(lua_State *L)
 {
 	// pos
 	v3s16 pos = read_v3s16(L, 1);
+	if (getClient(L)->getCSMFlavourLimits() & CSMFlavourLimit::CSM_FL_LOOKUP_NODES) {
+		// TODO
```

Comment:
So the server will specify an acceptable range? Thats good. I expect most servers will set a range of 0 because any range allows effectively 'seeing through walls'.